src/Silverback.Integration/Messaging/Validation/MessageValidator.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Silverback.Messaging.Validation
+{
+    /// <summary>
+    ///     Validates the message and throws a <see cref="MessageValidationException" /> depending on the
+    ///     <see cref="MessageValidationMode" />.
+    /// </summary>
+    internal static class MessageValidator
+    {
+        /// <summary>
+        ///     Validates the message.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="validationMode">The message validation mode.</param>
+        /// <param name="callback">The callback function to log the message validation errors.</param>
+        /// <exception cref="MessageValidationException">The thrown exception when validation mode is ThrowException.</exception>
+        public static void ValidateMessage(
+            object message,
+            MessageValidationMode validationMode,
+            Action<string> callback)
```

Comment:
Could use a meaningful name instead of `callback` (e.g. `logErrorCallback`).

But since it's not only a single case, you could just change the method signature to return a boolean indicating whether the validation succeded or failed and in the caller spot you just do `if (!MessageValidator.ValidateMessage(message, mode)) { // log }`. The other case is if it throws, but it doesn't matter since the rest of the caller code will be jumped.

If the signature is changed, the method name could also be changed in something like `CheckMessageIsValid` (or similar), which would make the return value more obvious (Is it valid> -> true or false).

Reply:
@BEagle1984 I removed the callback and refactored a little bit the method.